homeassistant/components/home_connect/light.py

```diff
@@ -0,0 +1,207 @@
+            self._key = COOKING_LIGHTING
+            self._custom_color_key = None
+            self._color_key = None
+
+    @property
+    def is_on(self):
+        """Return true if the light is on."""
+        return bool(self._state)
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def hs_color(self):
+        """Return the color property."""
+        return self._hs_color
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        if self._ambient:
+            return SUPPORT_BRIGHTNESS | SUPPORT_COLOR
+        return SUPPORT_BRIGHTNESS
+
+    async def async_turn_on(self, **kwargs):
+        """Switch the light on, change brightness, change color."""
+        if self._ambient:
+            if ATTR_BRIGHTNESS in kwargs or ATTR_HS_COLOR in kwargs:
+                try:
+                    await self.hass.async_add_executor_job(
+                        self.device.appliance.set_setting,
+                        self._color_key,
+                        BSH_AMBIENT_LIGHT_COLOR_CUSTOM_COLOR,
+                    )
+                except HomeConnectError as err:
+                    _LOGGER.error("Error while trying selecting customcolor: %s", err)
+                if self._brightness is not None:
+                    brightness = 10 + ceil(self._brightness / 255 * 90)
+                    if ATTR_BRIGHTNESS in kwargs:
+                        brightness = 10 + ceil(kwargs[ATTR_BRIGHTNESS] / 255 * 90)
+
+                    hs_color = kwargs.get(ATTR_HS_COLOR, default=self._hs_color)
+
+                    if hs_color is not None:
+                        rgb = color_util.color_hsv_to_RGB(*hs_color, brightness)
+                        hex_val = color_util.color_rgb_to_hex(rgb[0], rgb[1], rgb[2])
+                        try:
+                            await self.hass.async_add_executor_job(
+                                self.device.appliance.set_setting,
+                                self._custom_color_key,
+                                f"#{hex_val}",
+                            )
+                        except HomeConnectError as err:
+                            _LOGGER.error(
+                                "Error while trying setting the color: %s", err
+                            )
+            else:
+                _LOGGER.debug("Switching ambient light on for: %s", self.name)
+                try:
+                    await self.hass.async_add_executor_job(
+                        self.device.appliance.set_setting,
+                        self._key,
+                        True,
+                    )
+                except HomeConnectError as err:
+                    _LOGGER.error(
+                        "Error while trying to turn on ambient light: %s", err
+                    )
+                    self._state = False
```

Comment:
We're still changing state on error. I don't think we should do that. We can't guarantee that this is the correct current state.

Reply:
OK, I will remove the state setting after an error.

@DavidMStraub, for consistency reason we should probably create a PR to do the same for switch platform.